check_nightly_success/check-nightly-success/check.py

```diff
@@ -0,0 +1,74 @@
+import requests
+import itertools
+from datetime import datetime
+import sys
+import argparse
+import os
+
+# Constants
+GITHUB_TOKEN = os.environ["RAPIDS_GH_TOKEN"]
+GOOD_STATUSES = ("success",) # noqa
+
+
+def main(
+    repo: str,
+    repo_owner: str,
+    workflow_id: str,
+    max_days_without_success: int,
+):
+    headers = {"Authorization": f"token {GITHUB_TOKEN}"}
+    url = f"https://api.github.com/repos/{repo_owner}/{repo}/actions/workflows/{workflow_id}/runs"
+    response = requests.get(url, headers=headers)
+    response.raise_for_status()
+    runs = response.json()["workflow_runs"]
+    tz = datetime.fromisoformat(runs[0]["run_started_at"]).tzinfo
+    NOW = datetime.now(tz=tz)
+
+    branch_ok = {}
+    for branch, branch_runs in itertools.groupby(runs, key=lambda r: r["head_branch"]):
+        if branch.startswith("v"):  # Skip tags
```

Comment:
Is there a better way to do this? Maybe a different source of data to differentiate tags and branches?

Reply:
From offline discussions, Bradley's suggestion is to use one of the images from https://github.com/actions/runner-images, specifically probably 24.04, so I'm switching to that.

Reply:
I can match branch names instead.

Reply:
That's fine!